build_defs/sigh.bzl

```diff
@@ -0,0 +1,34 @@
+load(":run_in_repo.bzl", "EXECUTION_REQUIREMENTS_TAGS", "run_in_repo", "run_in_repo_test")
+
+def sigh_command(
+        name,
+        srcs,
+        sigh_cmd,
+        progress_message = "",
+        outs = [],
+        deps = [],
+        execute = True,
+        visibility = []):
+    """Runs the tool/sigh command from bazel with the given sign_cmd arguments.
+    Note: Any files (e.g. additional src, build outputs, etc) needed to be seen
+    by the sigh tool should be added in the deps attribute."""
+
+    run_macro = run_in_repo if execute else run_in_repo_test
+
+    run_macro(
+        name = name,
+        srcs = srcs,
+        outs = outs,
+        # TODO: generated header guard should contain whole workspace-relative
+        # path to file.
```

Comment:
This TODO is only relevant for schema2pkg (for C++). You can just delete it from here.